jupyverse_api/src/jupyverse_api/jupyterlab/__init__.py

```diff
@@ -17,29 +17,48 @@ def __init__(self, app: App, auth: Auth):
 
         @router.get("/lab")
         async def get_lab(
+            request: Request,
             user: User = Depends(auth.current_user()),
         ):
-            return await self.get_lab("lab", user)
+            try:
+                return await self.get_lab("lab", user, request)
+            except TypeError:
+                # For backward API compatibility
+                return await self.get_lab("lab", user)
```

Comment:
I'd rather use an FPS context:
```py
async with Context():
    put(request)
    return await self.get_lab("lab", user)
```
You can then get the request down the stack:
```py
with get_nowait(Request) as request:
    ...
```
You'll need FPS v0.5.7.